modules/cookbooks/pages/rag.adoc

```diff
@@ -0,0 +1,174 @@
+        title text NOT NULL,
+        body text NOT NULL,
+        embeddings vector(768) NOT NULL
+      );
+      CREATE INDEX IF NOT EXISTS text_hnsw_index
+        ON searchable_text 
+        USING hnsw (embeddings vector_l2_ops);
+    table: searchable_text
+    columns: ["id", "title", "body", "embeddings"]
+    args_mapping: "[this.article.id, this.article.title, this.article.content, this.article.embeddings.vector()]"
+----
+
+We can save this pipeline as `indexing.yaml` and run it with `rpk connect run ./indexing.yaml` to test and make sure our postgres table is being populated with embeddings.
+
+== Generating search responses
+
+In order to generate responses to questions over our dataset we're creating embeddings for, we'll use an xref:components:inputs/http_server.adoc[HTTP server input] to recieve the questions. Since we'll be running this pipeline in Streams mode, our HTTP server will xref:guides:streams_mode/about.adoc#http-endpoints[endpoints prefixed with stream identifier], which in our case is the same as the filename: `search`.
+
+[source,yaml]
+----
+input:
+  http_server:
+    path: /
+    allowed_verbs: [ GET ]
+    sync_response:
+      headers:
+        Content-Type: application/json
+----
+
+Our user query will come in as the query parameter `q` like so `http://localhost:4195/search?q=question_here`. Since query parameters as exposed as metadata in the `http_server` input, we can reference that in bloblang with `@q`.
+
+[source,yaml]
+----
+pipeline:
+  processors:
+    - label: compute_embeddings
+      ollama_embeddings:
+        model: nomic-embed-text
+        text: "search_query: %s".format(@q)
+----
+
+Now our payload is the embeddings vector and we can use Postgres to fetch the top 3 most similar documents to our embeddings vector using the xref:components:processors/sql_raw.adoc[`sql_raw`] processor.
+
+[source,yaml]
+----
+    - sql_raw:
+        driver: "postgres"
+        dsn: "${PG_CONNECTION_STRING}"
+        query: SELECT title, body FROM searchable_text ORDER BY embeddings <-> $1 LIMIT 3
+        args_mapping: "[ this.vector() ]"
+----
+
+With our looked up information, as well as our initial query, we can use the xref:components:processors/ollama_chat.adoc[`ollama_chat` processor] to respond to the user's question as text.
+
+[source,yaml]
+----
+    - label: generate_response
+      ollama_chat:
+        model: llama3.1
+        prompt: |
+          Your task is to respond to user queries using the provided information.
+
+          The user asked: ${! @q }
+          Context: ${!this.map_each(row -> "%s\n%s".format(row.title, row.body)).join("\n\n")}
+          Response:
+----
+
+Now that we've generated our response, we can send that back as the response to our HTTP server using xref:components:processors/sync_response.adoc[`sync_response`], then we will delete the message so nothing goes to the output using a xref:components:processors/mapping.adoc[bloblang mapping].
```

Comment:
```suggestion
Now that you've generated a response, you can send that back to the to our HTTP server as a response using xref:components:processors/sync_response.adoc[`sync_response`]. Then, delete the message using a xref:components:processors/mapping.adoc[bloblang mapping] so that nothing goes to the output.
```